// auto-generated by yaml_to_cxx.py from test/upstream/regression/2930.yaml
#include "testlib.h"
void test_upstream_regression_2930() {
    enter_section("test_upstream_regression_2930: Avoid misleading array limit error message");
    temp_table tbl_table;
    R::Query tbl = tbl_table.table();
    TEST_EQ(tbl.insert(R::range(1, 1000).map([=](R::Var i){ return R::object("id", *i, "mod", *i % 5, "foo", 5); })).pluck("first_error", "inserted").run(*conn), (R::Object{{"inserted",999}}));
    TEST_EQ(tbl.coerce_to("array").run(*conn), (err("RqlRuntimeError", "Array over size limit `500`.", R::Array{0})));
    TEST_EQ(tbl.group("mod").coerce_to("array").run(*conn), (err("RqlRuntimeError", "Grouped data over size limit `500`.  Try putting a reduction (like `.reduce` or `.count`) on the end.", R::Array{0})));
    TEST_EQ(tbl.group("foo").coerce_to("array").run(*conn), (err("RqlRuntimeError", "Grouped data over size limit `500`.  Try putting a reduction (like `.reduce` or `.count`) on the end.", R::Array{0})));
    exit_section();
}
